lib/internal/test_runner/test.js

```diff
@@ -171,18 +173,18 @@ class Test extends AsyncResource {
 
     if (parent === null) {
       this.concurrency = 1;
-      this.indent = '';
+      this.nesting = 0;
       this.only = testOnlyFlag;
-      this.reporter = new TapStream();
+      this.reporter = hasReporters ? new ReporterStream() : new TapStream();
```

Comment:
This always needs to be `ReporterStream()` now.

Reply:
@cjihrig why? we still need to report tap to the primary process running "--test"

Reply:
I think we should have a single interface for reporting. `ReporterStream` can be that interface. If the user does not specify a reporter (or in the case `--test` is used), then `TapStream` should be the reporter and listen for data from `ReporterStream`.